presto-main/src/main/java/com/facebook/presto/cost/ExchangeStatsRule.java

```diff
@@ -49,9 +52,11 @@ public ExchangeStatsRule(StatsNormalizer normalizer)
     protected Optional<PlanNodeStatsEstimate> doCalculate(ExchangeNode node, StatsProvider statsProvider, Lookup lookup, Session session, TypeProvider types)
     {
         Optional<PlanNodeStatsEstimate> estimate = Optional.empty();
+        double totalSize = 0;
         for (int i = 0; i < node.getSources().size(); i++) {
             PlanNode source = node.getSources().get(i);
             PlanNodeStatsEstimate sourceStats = statsProvider.getStats(source);
+            totalSize = (!isNaN(sourceStats.getOutputSizeInBytes()) && !isNaN(totalSize)) ? totalSize + sourceStats.getOutputSizeInBytes() : NaN;
```

Comment:
nit: it can be safely simplified to `totalSize += sourceStats.getOutputSizeInBytes()` (anything + NaN is NaN)